db/mongo/client.go

```diff
@@ -0,0 +1,207 @@
+package mongo
+
+import (
+	"ApniUniversity/config"
+	"ApniUniversity/db"
+	"ApniUniversity/models"
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"time"
+)
+
+const (
+	teacherCollection = "Teacher"
+	subjectCollection = "Subject"
+)
+
+type client struct {
+	connection *mongo.Client
+}
+
+func NewClient(_ db.Options) (db.DataStore, error) {
+	uri := fmt.Sprintf("mongodb://%s:%s", viper.GetString(config.DbHost), viper.GetString(config.DbPort))
+	log().Infof("Initializing MongoDB At: %s", uri)
```

Comment:
log should be at the top

Reply:
Sir log is using uri in its formatted string